site/content/en/latest/user/private-key-provider.md

```diff
@@ -0,0 +1,306 @@
+---
+title: "Using Envoy Private Key Provider for TLS termination"
+---
+
+To allow the accelerate the private key operation through hardware or protect the private key through the hardware, the [Envoy Private Key Provider](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-privatekeyprovider) is added to Envoy.
+
+Today, there are two private key providers were implemented as Envoy contrib-extension:
+- [QAT in Envoy 1.24 release](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/private_key_providers/qat/v3alpha/qat.proto#extensions-private-key-providers-qat-v3alpha-qatprivatekeymethodconfig)
+- [CryptoMB in Envoy 1.20 release](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/private_key_providers/cryptomb/v3alpha/cryptomb.proto )
+
+Both of them are used to accelerate the TLS handshake through the hardware capabilities.
+
+This guide will walk through the steps required to configure TLS Terminate mode for TCP traffic and same time using the Envoy Private Key Provider to accelerate the TLS handshake.
+
+## Prerequisites
+
+### For QAT
+
+- Install Linux kernel 5.17 or similar
+- Ensure the node has QAT devices by checking the QAT physical function devices presented.
+
+  ```shell
+  lspci -d 8086:4940
+  ```
+
+- Enable IOMMU from BIOS
+- Enable IOMMU for Linux kernel
+
+  ```shell
+  cat /etc/default/grub:
+  GRUB_CMDLINE_LINUX="intel_iommu=on vfio-pci.ids=8086:4941"
+  update-grub
+  reboot
+  ````
+
+  Once the system is rebooted, check if the IOMMU has been enabled via the following command:
+
+  ```shell
+  dmesg| grep IOMMU
+  [    1.528237] DMAR: IOMMU enabled
+  ```
+
+- Enable virtual function devices for QAT device
+
+  ```shell
+  rmmod qat_4xxx
+  modprobe qat_4xxx
+  qat_device=$(lspci -D -d :4940 | awk '{print $1}')
+  for i in $qat_device; do echo 16|sudo tee /sys/bus/pci/devices/$i/sriov_numvfs; done
+  ```
+
+- Enhance the container runtime memory lock limit
+
+  ```shell
+  mkdir /etc/systemd/system/containerd.service.d
+  cat <<EOF >>/etc/systemd/system/containerd.service.d/memlock.conf
+  [Service]
+  LimitMEMLOCK=134217728
+  EOF
+  ```
+
+  Restart the container runtime (for containerd, CRIO has similar concept)
+
+  ```shell
+  systemctl daemon-reload
+  systemctl restart containerd
+  ```
+
+- Install [Intel® QAT Device Plugin for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes)
+
+  ```shell
+  kubectl apply -k 'https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/qat_plugin?ref=main'
+  ```
+
+  Verification of the plugin deployment and detection of QAT hardware can be confirmed by examining the resource allocations on the nodes:
+
+  ```shell
+  kubectl get node -o yaml| grep qat.intel.com
+  ```
+
+### For CryptoMB:
+
+It required the node with 3rd generation Intel Xeon Scalable processor server processors, or later.
```

Comment:
lets add the o/p from the commands so a user is certain their platform can support this feature

Reply:
ok, got it.

Reply:
I found I already added in the below